include/onnxruntime/core/providers/dml/dml_provider_factory.h

```diff
@@ -44,6 +44,52 @@ ORT_API_STATUS(OrtSessionOptionsAppendExecutionProvider_DML, _In_ OrtSessionOpti
 ORT_API_STATUS(OrtSessionOptionsAppendExecutionProviderEx_DML, _In_ OrtSessionOptions* options,
                _In_ IDMLDevice* dml_device, _In_ ID3D12CommandQueue* cmd_queue);
 
+
+struct OrtDmlApi;
+typedef struct OrtDmlApi OrtDmlApi;
+
+const OrtDmlApi* GetOrtDmlApi(_In_ uint32_t version) NO_EXCEPTION;
```

Comment:
>GetOrtDmlApi [](http://example.com/codeflow?start=17&length=12)

Do we need/want this free function? Don't we want people to call `GetExecutionProviderApi` instead?

If this is only internal, for `GetExecutionProviderApi` to call, then shouldn't we put this function somewhere more private? Alternately (per other comment) put the public interface `OrtDmlApi` in another header file.

Reply:
Moved into onnxruntime_c_api.cc under the USE_DML macro (which is the only place it is used)